gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/service/granter/code/AuthorizationCodeTokenGranter.java

```diff
@@ -158,6 +159,10 @@ private void checkPKCE(TokenRequest tokenRequest, AuthorizationCode authorizatio
         String codeChallenge = parameters.getFirst(Parameters.CODE_CHALLENGE);
         String codeChallengeMethod = parameters.getFirst(Parameters.CODE_CHALLENGE_METHOD);
 
+        if (codeChallenge == null && authorizationCode.getRequestParameters() != null && authorizationCode.getRequestParameters().containsKey(Parameters.CODE_CHALLENGE)) {
+            throw new InvalidGrantException("Missing parameter: code_challenge");
+        }
+
```

Comment:
codeChallenge is already coming from the authorizationCode.getRequestParameters()